app/src/main/java/com/infomaniak/drive/ui/home/RootFilesFragment.kt

```diff
@@ -82,46 +89,96 @@ class RootFilesFragment : Fragment() {
         )
 
         setupRootPendingFilesIndicator(mainViewModel.pendingUploadsCount, rootFilesUploadFileInProgressView)
+
+        if (fileListViewModel.justLaunched.not()) {
+            uiSettings.lastVisitedRootFileTreeCategory = null
+        }
+        if (fileListViewModel.justLaunched) lifecycleScope.launch {
```

Comment:
You should extract all the `lifecycleScope.launch` in its own function
The onCreateView method is already big, and this launch is really consequent

Reply:
Also it will permit you to remove the null clause of the `when`because the compilator won't think you return something

Reply:
> Also it will permit you to remove the null clause of the whenbecause the compilator won't think you return something

The compiler no longer tolerates non exhaustive when expressions when the subject type is an enum, a sealed class, or a sealed interface, so I'll need to keep that when branch.

Reply:
Thank you, I extracted all that logic to a new function. I agree it's more digest that way.